Macro name built by ## isn't expanded when called through another macro

repro.h:

```c
#define CONCAT_(a, b) a##b
#define CONCAT(a, b)  CONCAT_(a, b)
#define APPLY(macro, args) macro args

#define OVERLOAD1(x) x

int call_via_apply = APPLY(CONCAT(OVERLOAD, 1), (7));
```

```
$ gcc -E -P repro.h

int call_via_apply = 7;

$  clang -E -P repro.h
int call_via_apply = 7;

$ arocc -E -P repro.h
int call_via_apply = OVERLOAD1  (7);
```

Resulting in an error:

```
$ arocc repro.h
repro.h:7:22: error: call to undeclared function 'OVERLOAD1'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
int call_via_apply = APPLY(CONCAT(OVERLOAD, 1), (7));
                     ^
repro.h:3:28: note: expanded from here
#define APPLY(macro, args) macro args
                           ^
repro.h:7:28: note: expanded from here
int call_via_apply = APPLY(CONCAT(OVERLOAD, 1), (7));
                           ^
repro.h:2:23: note: expanded from here
#define CONCAT(a, b)  CONCAT_(a, b)
                      ^
note: (skipping 2 expansions in backtrace; use -fmacro-backtrace-limit=0 to see all)
repro.h:2:31: note: expanded from here
#define CONCAT(a, b)  CONCAT_(a, b)
                              ^
repro.h:7:35: note: expanded from here
int call_via_apply = APPLY(CONCAT(OVERLOAD, 1), (7));
                                  ^
<scratch space>:1:1: note: expanded from here
OVERLOAD1
^
repro.h:7:22: warning: passing arguments to a function without a prototype is deprecated in all versions of C and is not supported in C23 [-Wdeprecated-non-prototype]
int call_via_apply = APPLY(CONCAT(OVERLOAD, 1), (7));
                     ^
repro.h:3:34: note: expanded from here
#define APPLY(macro, args) macro args
                                 ^
repro.h:7:50: note: expanded from here
int call_via_apply = APPLY(CONCAT(OVERLOAD, 1), (7));
                                                 ^
1 warning and 1 error generated.
```

--- a/src/aro/Preprocessor.zig
+++ b/src/aro/Preprocessor.zig
@@ -2995,7 +2995,7 @@ fn pasteTokens(pp: *Preprocessor, lhs_toks: *ExpandBuf, rhs_toks: []const TokenW
         pasted_token.id;
 
     const hideset = try pp.hideset.intersection(pp.hideset.get(lhs.loc), pp.hideset.get(rhs.loc));
-    const generated_token = try pp.makeGeneratedTokenExtra(start, pasted_id, lhs, hideset);
+    const generated_token = try pp.makeGeneratedTokenExtra(start, pasted_id, lhs, hideset, lhs.flags.is_macro_arg or rhs.flags.is_macro_arg);
     try lhs_toks.append(gpa, generated_token);
 
     if (next.id != .nl and next.id != .eof) {
@@ -3006,13 +3006,13 @@ fn pasteTokens(pp: *Preprocessor, lhs_toks: *ExpandBuf, rhs_toks: []const TokenW
     try bufCopyTokens(gpa, lhs_toks, rhs_toks[rhs_rest..], &.{});
 }
 
-fn makeGeneratedTokenExtra(pp: *Preprocessor, start: usize, id: Token.Id, source: TokenWithExpansionLocs, hideset: Hideset.Index) !TokenWithExpansionLocs {
+fn makeGeneratedTokenExtra(pp: *Preprocessor, start: usize, id: Token.Id, source: TokenWithExpansionLocs, hideset: Hideset.Index, is_macro_arg: bool) !TokenWithExpansionLocs {
     const gpa = pp.comp.gpa;
     var pasted_token = TokenWithExpansionLocs{ .id = id, .loc = .{
         .id = .generated,
         .byte_offset = @intCast(start),
         .line = pp.generated_line,
-    } };
+    }, .flags = .{ .is_macro_arg = is_macro_arg } };
     pp.generated_line += 1;
     try pasted_token.addExpansionLocation(gpa, &.{source.loc});
     try pasted_token.addExpansionLocation(gpa, source.expansionSlice());
@@ -3022,7 +3022,7 @@ fn makeGeneratedTokenExtra(pp: *Preprocessor, start: usize, id: Token.Id, source
 }
 
 fn makeGeneratedToken(pp: *Preprocessor, start: usize, id: Token.Id, source: TokenWithExpansionLocs) !TokenWithExpansionLocs {
-    return pp.makeGeneratedTokenExtra(start, id, source, pp.hideset.get(source.loc));
+    return pp.makeGeneratedTokenExtra(start, id, source, pp.hideset.get(source.loc), false);
 }
 
 /// Defines a new macro and warns if it is a duplicate

--- a/test/cases/apply pasted macro name.c
+++ b/test/cases/apply pasted macro name.c
@@ -1,0 +1,11 @@
+#define CONCAT_(a, b) a##b
+#define CONCAT(a, b)  CONCAT_(a, b)
+#define APPLY(macro, args) macro args
+
+#define OVERLOAD1(x) x
+
+int call_via_apply = APPLY(CONCAT(OVERLOAD, 1), (7));
+
+/** manifest:
+expand
+*/

--- a/test/cases/expanded/apply pasted macro name.c
+++ b/test/cases/expanded/apply pasted macro name.c
@@ -1,0 +1,1 @@
+int call_via_apply = 7;

